packages/react/src/components/Forms/Select/mono-searchable.tsx

```diff
@@ -52,6 +66,13 @@ export const SelectMonoSearchable = (props: SubProps) => {
     downshiftReturn.selectItem(optionToSelect ?? null);
   }, [props.value, props.options, props.downshiftProps]);
 
+  // Even there is already a value selected, when opening the combobox menu we want to display all available choices.
+  useEffect(() => {
+    if (downshiftReturn.isOpen) {
+      setOptionsToDisplay(props.options);
+    }
+  }, [downshiftReturn.isOpen]);
+
```

Comment:
```suggestion
  }, [downshiftReturn.isOpen, props.options]);
```

Reply:
Doing this will cause:
-> If options change when typing for filtering it will display all options totally ignoring the filter. Which is a non-sense

WDYT?

Reply:
If the options change when the select is open, the new options will not pass in `setOptionsToDisplay`, so actually if the user clicks on one of the option proposed, it well send back in `onChange` an old value who can be not part of the new option. Depend the case it could potentially create a bug.

Reply:
If the options change when the select is open, the new options will not pass in `setOptionsToDisplay`, so actually if the user clicks on one of the option proposed, it well send back in `onChange` an old value who can be not part of the new option. Depend the case it could potentially create a bug.

Reply:
imo this is out the scope of this useEffect which is role is "Display all options only once when isOpen turns true". Adding here this new dep would result in the bug I described. 

Regarding your use case we could do another PR to treat this potential issue which needs to be done in another part of the code, because this was existing before this PR.

In reality imo we cannot always apply the "add all used variables to the deps", in most cases the called methods use lots of other hidden variables that we never add as deps and that's ok. Sometimes it could cause bugs like the one I described.

Reply:
It would have been nice to fix directly these 2 issues when we could do it directly:

> -> If options change when typing for filtering it will display all options totally ignoring the filter.

> If the options change when the select is open, the new options will not pass in setOptionsToDisplay, so actually if the user clicks on one of the option proposed, it will send back in onChange an old value who could be not part of the new options. Depend the case it could potentially create a bug.

But let's create another PR if you feel more confortable.

Reply:
Yes I want to be able to make a release in between 😬